validator/coordinator/src/main/java/tech/pegasys/teku/validator/coordinator/ValidatorApiHandler.java

```diff
@@ -471,15 +480,17 @@ private ProposerDuties getProposerDutiesFromIndexesAndState(
               }
               result.add(new ProposerDuty(publicKey, maybeIndex.get(), slot));
             });
-    return new ProposerDuties(getCurrentDutyDependentRoot(state), result);
+    return new ProposerDuties(
+        specProvider.atEpoch(epoch).getBeaconStateUtil().getCurrentDutyDependentRoot(state),
+        result);
   }
 
   private AttesterDuties getAttesterDutiesFromIndexesAndState(
       final BeaconState state, final UInt64 epoch, final Collection<Integer> validatorIndexes) {
     final Bytes32 dependentRoot =
-        epoch.isGreaterThan(get_current_epoch(state))
-            ? getCurrentDutyDependentRoot(state)
-            : getPreviousDutyDependentRoot(state);
+        epoch.isGreaterThan(specProvider.getCurrentEpoch(state))
+            ? specProvider.at(state).getBeaconStateUtil().getCurrentDutyDependentRoot(state)
+            : specProvider.at(state).getBeaconStateUtil().getPreviousDutyDependentRoot(state);
```

Comment:
`state` could be from the previous epoch (because it only has to be within the look ahead period). I suspect we should explicitly ask for the spec at `epoch`.